src/resolvers/projectResolver.ts

```diff
@@ -806,136 +907,37 @@ export class ProjectResolver {
-        projectUpdateId: update.id,
-        reaction,
-      });
-
-      project.updateQualityScoreHeart(true);
-      project.save();
-
-      await Reaction.save(newReaction);
-    }
-    await updateTotalReactionsOfAProject(update.projectId);
-
-    return true;
-  }
-
-  @Mutation(returns => ToggleResponse)
-  async toggleProjectReaction(
-    @Arg('projectId') projectId: number,
-    @Arg('reaction') reaction: REACTION_TYPE = 'heart',
+  @Query(returns => [ProjectUpdate])
+  async getProjectUpdates(
+    @Arg('projectId', type => Int) projectId: number,
+    @Arg('skip', type => Int, { defaultValue: 0 }) skip: number,
+    @Arg('take', type => Int, { defaultValue: 10 }) take: number,
+    @Arg('connectedWalletUserId', type => Int, { nullable: true })
+    connectedWalletUserId: number,
     @Ctx() { req: { user } }: MyContext,
-  ): Promise<object> {
-    if (!user) throw new Error(errorMessages.AUTHENTICATION_REQUIRED);
-
-    const project = await Project.findOne({ id: projectId });
-
-    if (!project) throw new Error(errorMessages.PROJECT_NOT_FOUND);
-
-    let update = await ProjectUpdate.findOne({ projectId, isMain: true });
-    if (!update) {
-      update = await ProjectUpdate.save(
-        await ProjectUpdate.create({
-          userId:
-            project && project.admin && +project.admin ? +project.admin : 0,
+  ): Promise<ProjectUpdate[]> {
+    let query = this.projectUpdateRepository
+      .createQueryBuilder('projectUpdate')
+      .where(
+        'projectUpdate.projectId = :projectId and projectUpdate.isMain = false',
+        {
           projectId,
-          content: '',
-          title: '',
-          createdAt: new Date(),
-          isMain: true,
-        }),
-      );
-    }
-
-    const usersReaction = await Reaction.findOne({
-      projectUpdateId: update.id,
-      userId: user.userId,
-    });
-    const [, reactionCount] = await Reaction.findAndCount({
-      projectUpdateId: update.id,
-    });
-
-    await Reaction.delete({ projectUpdateId: update.id, userId: user.userId });
-    const response = new ToggleResponse();
-    response.reactionCount = reactionCount;
-
-    if (usersReaction && usersReaction.reaction === reaction) {
-      response.reaction = false;
-      response.reactionCount = response.reactionCount - 1;
-    } else {
-      const newReaction = await Reaction.create({
-        userId: user.userId,
-        projectUpdateId: update.id,
-        reaction,
-        project,
-      });
+        },
+      )
+      .take(take)
+      .skip(skip);
 
-      await Reaction.save(newReaction);
-      response.reactionCount = response.reactionCount + 1;
-      response.reaction = true;
+    const viewerUserId = connectedWalletUserId || user?.userId;
+    if (viewerUserId) {
+      query = query.leftJoinAndMapOne(
+        'projectUpdate.reaction',
+        Reaction,
+        'reaction',
+        'reaction.projectUpdateId = CAST(projectUpdate.id AS INTEGER) AND reaction.userId = :viewerUserId',
```

Comment:
Is it required to CAST as INTEGER?
Both columns are type ID and js wise as Number, maybe the casting isn't required.